internal/services/cdn/cdn_endpoint_custom_domain_resource_test.go

```diff
@@ -80,6 +111,76 @@ func TestAccCdnEndpointCustomDomain_requiresImport(t *testing.T) {
 	})
 }
 
+func TestAccCdnEndpointCustomDomain_httpsCdn(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_cdn_endpoint_custom_domain", "test")
+
+	preCheck(t)
+
+	r := NewCdnEndpointCustomDomainResource(os.Getenv("ARM_TEST_DNS_ZONE_RESOURCE_GROUP_NAME"), os.Getenv("ARM_TEST_DNS_ZONE_NAME"))
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.httpsCdn(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccCdnEndpointCustomDomain_httpsCdnUpdate(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_cdn_endpoint_custom_domain", "test")
+
+	preCheck(t)
+
+	r := NewCdnEndpointCustomDomainResource(os.Getenv("ARM_TEST_DNS_ZONE_RESOURCE_GROUP_NAME"), os.Getenv("ARM_TEST_DNS_ZONE_NAME"))
```

Comment:
_minor_ but could we move the pre-check method onto the Resource here:

```suggestion
	r := NewCdnEndpointCustomDomainResource(os.Getenv("ARM_TEST_DNS_ZONE_RESOURCE_GROUP_NAME"), os.Getenv("ARM_TEST_DNS_ZONE_NAME"))
	r.preCheck(t)
```

since this is scoped to the Endpoint resource and 'preCheck' is a pretty generic name

Reply:
Indeed